cmd/ausoceantv/src/plan-element.ts

```diff
@@ -0,0 +1,60 @@
+import { html } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import { TailwindElement } from "./shared/tailwind.element.ts";
+
+@customElement("plan-element")
+export class planElement extends TailwindElement() {
+  // Type of plan: eg 'Day Pass' or 'Monthly Subscription'.
+  @property({ type: String, attribute: "plan-type" })
+  planType = "";
+
+  // URL to redirect to when plan is selected.
+  @property({ type: String, attribute: "href" })
+  href = "";
+
+  // Cost of the plan in dollars.
+  @property({ type: Number, attribute: "plan-cost" })
+  planCost = "";
+
+  // Comparison cost of paying for a different plan in dollars.
+  @property({ type: Number, attribute: "comp-cost" })
+  compCost = "";
+
+  firstUpdated() {
+    this.addCompPrice();
+  }
+
+  render() {
+    return html`
+      <div class="bg-gray-300 rounded h-fit flex flex-col gap-5 overflow-hidden items-center">
+        <p class="text-lg bg-gray-800 text-white p-5 font-bold w-full text-center">${this.planType}</p>
+        <div id="price" class="flex justify-center">
+          <p class="text-3xl font-bold">$${this.planCost}</p>
+        </div>
+        <button @click="${this.selectPlan}" class="bg-gray-800 text-white w-fit px-10 py-2 rounded mb-5">Select Plan</button>
+      </div>
+    `;
+  }
+
+  // Used to add a strikethrough price to compare costs.
+  addCompPrice() {
```

Comment:
This can be simplified somewhat by using lit's html templating. Then you can get rid of firstUpdated and addCompPrice.

```ts
render() {
    let comparePrice = html``
    if (this.compCost != "") {
        comparePrice = html`<p class="line-through">$${this.compCost}</p>`
    }

    return html`
      <div class="bg-gray-300 rounded h-fit flex flex-col gap-5 overflow-hidden items-center">
        <p class="text-lg bg-gray-800 text-white p-5 font-bold w-full text-center">${this.planType}</p>
        <div id="price" class="flex justify-center">
          <p class="text-3xl font-bold">$${this.planCost}</p>
          ${comparePrice}
        </div>
        <button @click="${this.selectPlan}" class="bg-gray-800 text-white w-fit px-10 py-2 rounded mb-5">Select Plan</button>
      </div>
    `;
}

Reply:
I knew there must be an easier way. Thanks :+1: